src/actions/move.cpp

```diff
@@ -138,25 +138,32 @@ void move_unit_spectator::set_unit(const unit_map::const_iterator &u)
 
 game_events::pump_result_t get_village(const map_location& loc, int side, bool *action_timebonus, bool fire_event)
 {
-	std::vector<team> &teams = resources::gameboard->teams();
-	team *t = static_cast<unsigned>(side - 1) < teams.size() ? &teams[side - 1] : nullptr;
-	if (t && t->owns_village(loc)) {
-		return game_events::pump_result_t();
-	}
+	team* t = nullptr;
+	bool not_defeated = false;
 
-	bool not_defeated = t && !resources::gameboard->team_is_defeated(*t);
+	for(team& tm : resources::gameboard->teams()) {
+		if(tm.side() == side) {
+			if(tm.owns_village(loc)) {
+				return {};
+			} else {
+				not_defeated = !resources::gameboard->team_is_defeated(tm);
+				t = &tm;
+				break;
+			}
+		}
+	}
```

Comment:
replacing an index access with a "loop and find the correct index" seems a bit odd to me, maybe you coudl just add a get_team or get_optional_team function to game_board that takes a 1-based index.

Reply:
Rolled these loops back. I can address the indexing separately.